modules/wpml/wpml-config.php

```diff
@@ -281,23 +403,60 @@ protected function register_or_translate_option( $context, $name, $key ) {
 	 * Recursively transforms xml nodes to an array, ready for PLL_Translate_Option.
 	 *
 	 * @since 2.9
+	 * @since 3.3 Type-hinted the parameters `$key` and `$arr`.
+	 * @since 3.3 Added the parameter `$fill_value`.
 	 *
-	 * @param SimpleXMLElement $key XML node.
-	 * @param array            $arr Array of option keys to translate.
+	 * @param SimpleXMLElement $key        XML node.
+	 * @param array            $arr        Array of option keys to translate.
+	 * @param mixed            $fill_value Value to use when filling entries. Default is true.
 	 * @return array
 	 */
-	protected function xml_to_array( $key, &$arr = array() ) {
+	protected function xml_to_array( SimpleXMLElement $key, array &$arr = array(), $fill_value = true ) {
```

Comment:
`$arr` is passed by reference even if `xml_to_array()` returns the modified array. Is it on purpose?
Wouldn't it be better to avoid passing by ref?